xanmod-kernels: add TT variant as dev

cassandracomar opened the pull request:
###### Description of changes

the xanmod-tt kernel used to be the default but was removed in #165257. this PR adds back the xanmod1-tt stable variant as linux_xanmod_dev, leaving the default linux_xanmod kernel to the base stable variant. note that the TT scheduler patches have not been made available for the edge variant upstream so we can't add a matching linux_xanmod_latest_dev variant.

alternatives considered:
* just use overrides to set the TT variant: this patched version is published as a separate release so this can't be easily overridden by users.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Commit:
xanmod-kernels: add stableTT variant

the xanmod-tt kernel used to be the default but was removed in #165257.
this commit adds back the xanmod1-tt stable variant as linux_xanmod_tt,
leaving the default linux_xanmod kernel to the base stable variant. note
that the TT scheduler patches have not been made available for the edge
variant upstream so we can't add a matching linux_xanmod_latest_tt
variant.

fortuneteller2k commented on pkgs/os-specific/linux/kernel/xanmod-kernels.nix:
```diff
@@ -7,6 +7,12 @@ let
     hash = "sha256-MeH9RUPDiuN22eAZ18v+N3aIT18dQ3FnTkcQV0MjB4k=";
   };
 
+  stableTTVariant = {
```
```suggestion
  devVariant = {
```

fortuneteller2k replied:
It isn't guaranteed that it will be TT forever. So I'm leaning towards naming it `dev` instead since it's under development.

fortuneteller2k commented on pkgs/os-specific/linux/kernel/xanmod-kernels.nix:
```diff
@@ -83,5 +89,6 @@ let
 in
 {
   stable = xanmodKernelFor stableVariant;
+  stableTT = xanmodKernelFor stableTTVariant;
```
```suggestion
  dev = xanmodKernelFor devVariant;
```

fortuneteller2k commented on pkgs/top-level/all-packages.nix:
```diff
@@ -23420,6 +23420,8 @@ with pkgs;
   # XanMod kernel
   linuxPackages_xanmod = linuxKernel.packages.linux_xanmod;
   linux_xanmod = linuxKernel.kernels.linux_xanmod;
+  linuxPackages_xanmod_tt = linuxKernel.packages.linux_xanmod_tt;
+  linux_xanmod_tt = linuxKernel.kernels.linux_xanmod_tt;
```
```suggestion
  linuxPackages_xanmod_dev = linuxKernel.packages.linux_xanmod_dev;
  linux_xanmod_dev = linuxKernel.kernels.linux_xanmod_dev;
```

fortuneteller2k commented on pkgs/top-level/linux-kernels.nix:
```diff
@@ -225,6 +225,13 @@ in {
       ];
     }).stable;
 
+    linux_xanmod_tt = (xanmodKernels {
+      kernelPatches = [
+        kernelPatches.bridge_stp_helper
+        kernelPatches.request_key_helper
+      ];
+    }).stableTT;
```
```suggestion
    linux_xanmod_dev = (xanmodKernels {
      kernelPatches = [
        kernelPatches.bridge_stp_helper
        kernelPatches.request_key_helper
      ];
    }).dev;
```

fortuneteller2k commented on pkgs/top-level/linux-kernels.nix:
```diff
@@ -561,6 +568,7 @@ in {
     linux_zen = recurseIntoAttrs (packagesFor kernels.linux_zen);
     linux_lqx = recurseIntoAttrs (packagesFor kernels.linux_lqx);
     linux_xanmod = recurseIntoAttrs (packagesFor kernels.linux_xanmod);
+    linux_xanmod_tt = recurseIntoAttrs (packagesFor kernels.linux_xanmod_tt);
```
```suggestion
    linux_xanmod_dev = recurseIntoAttrs (packagesFor kernels.linux_xanmod_dev);
```

Commit:
xanmod-kernels: rename TT to devVariant

Co-authored-by: fortune <lythe1107@gmail.com>

fortuneteller2k commented:
@ofborg build linux_xanmod_dev

cassandracomar commented:
> error: unused option: LRNG

I can't find this option in the CONFIG for the base stable or edge variants -- does this test pass for those variants?

fortuneteller2k commented:
> > error: unused option: LRNG
> 
> I can't find this option in the CONFIG for the base stable or edge variants -- does this test pass for those variants?

Only for base stable, I think we should just outright remove it now since it's broken for the majority of the variants.

cassandracomar commented:
looks like it's been removed from base stable as well: https://github.com/xanmod/linux/commit/181bccbacc4fef6f67cc6cbf3441a86df3f32481.

let me update the version of stable and edge so removing the option makes complete sense.

Commit:
xanmod-kernels: remove LRNG option & bump versions

this option was removed from stable in 5.15.45:
https://github.com/xanmod/linux/commit/181bccbacc4fef6f67cc6cbf3441a86df3f32481

and it's long been gone from edge.

cassandracomar commented:
@ofborg build linux_xanmod linux_xanmod_dev linux_xanmod_latest

cassandracomar commented:
@ofborg build linux_xanmod

cassandracomar commented:
@ofborg build linux_xanmod_dev

cassandracomar commented:
@ofborg build linux_xanmod_latest

cassandracomar commented:
ok @fortuneteller2k, all the builds have passed. I think this is ready for review.

Atemu reviewed (changes requested):
IMO `dev` is ambiguous and sounds like a general development branch from which the regular xanmod kernels are later spun off of.   
If the name changes later, we will simply follow that. Users would have to figure out that the name on the official site `_tt` somehow relates to our `_dev` otherwise.

Atemu commented:
Created https://github.com/NixOS/nixpkgs/pull/178165; a tree I've been using for weeks.

cassandracomar commented:
sorry was out of commission for a while. will update this PR this evening.

Commit:
revert name to tt

Commit:
update to 5.15.54

fortuneteller2k commented:
@ofborg build linux_xanmod linux_xanmod_tt linux_xanmod_latest
